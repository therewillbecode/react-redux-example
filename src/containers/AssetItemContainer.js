import React, { Component } from "react";
import { connect } from "react-redux";

import AssetItem from "../components/AssetItem";
import { findAssetById } from "../selectors/index";

class AssetItemContainer extends Component {
  render() {
    const { asset } = this.props;
    return <AssetItem asset={asset} />;
  }
}

const mapStateToProps = (state, ownProps) => ({
  asset: findAssetById(state.assets, ownProps.match.params.id)
});

export default connect(mapStateToProps)(AssetItemContainer);
