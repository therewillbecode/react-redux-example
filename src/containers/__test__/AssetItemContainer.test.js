/* eslint-disable */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";

import { AssetItemContainer } from "../AssetItemContainer";

describe("AssetItemContainer", () => {
  it("renders correctly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<AssetItemContainer />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
