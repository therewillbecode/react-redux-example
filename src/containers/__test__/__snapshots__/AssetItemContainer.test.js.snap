// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetItemContainer renders correctly 1`] = `
<AssetItem
  asset={undefined}
/>
`;
